crates/proof-of-sql/src/base/database/column_operation.rs

```diff
@@ -398,6 +505,63 @@ where
     ))
 }
 
+/// Divide an owned column by another.
+///
+/// Notes:
+/// 1. We do not check for length equality here.
+/// 2. We use floor division for rounding.
+/// 3. If division by zero occurs, we return an error.
+/// 4. Precision and scale follow T-SQL rules. That is,
+///   - new_scale = max(6, right_precision + left_scale + 1)
+///   - new_precision = left_precision - left_scale + right_scale + new_scale
+pub(crate) fn try_divide_decimal_columns<S, T0, T1>(
+    lhs: &[T0],
+    rhs: &[T1],
+    left_column_type: ColumnType,
+    right_column_type: ColumnType,
+) -> ColumnOperationResult<(Precision, i8, Vec<S>)>
+where
+    S: Scalar + core::convert::From<T0> + core::convert::From<T1>,
```

Comment:
This would result in bypassing `Scalar` conversions if `lhs` and `rhs` are integers:
```suggestion
    S: Scalar,
    T0: Into<BigInt>,
    T1: Into<BigInt>,
```

Reply:
@JayWhite2357 If they are both integers we shouldn't even use this function.

Reply:
But if one of them are, we do, right?